### ISSUE TITLE
No providers available

### ISSUE THREAD

**Issue by MeggyPoe** [Joniff/Terratype#1]
@Joniff  I have similar issues as  this one. https://our.umbraco.org/projects/backoffice-extensions/terratype/bugs-questions/83810-no-providers-available

On localhost everything is working, on production I get same error as she did. I tried installing it via Nuget and) via Backoffice, none did solve the problem. Umbraco version is 7.5.8.

**Comment by Joniff** [comment]
I guess you are running version 1.0.3 on production server.

If its possible, and I can understand it means pushing new code up to production.

But could you run the following code and 

 ```
              Dictionary<string, Type> installed = new Dictionary<string, Type>();
                Type baseType = typeof(Provider);
                foreach (Assembly currAssembly in AppDomain.CurrentDomain.GetAssemblies())
                {
                    Type[] typesInAsm;
                    try
                    {
                        typesInAsm = currAssembly.GetTypes();
                    }
                    catch (ReflectionTypeLoadException ex)
                    {
                        typesInAsm = ex.Types;
                    }

                    foreach (Type type in typesInAsm)
                    {
                        if (!type.IsClass || type.IsAbstract ||
                            !type.IsSubclassOf(baseType))
                        {
                            continue;
                        }

                        var derivedObject = System.Activator.CreateInstance(type) as Provider;
                        if (derivedObject != null)
                        {
                            installed.Add(derivedObject.Id, derivedObject.GetType());
                        }
                    }
                }
                return installed;
```

and somehow log the return value. Its the code, that I'm using to find 'providers'

If you can that would be great, If you can't I'll see what I can find to see why this code would be failing.

Cheers

**Comment by MeggyPoe** [comment]
Hm on localhost It works and I get: Terratype.GoogleMapsV3, Terratype.Providers.GoogleMapsV3]
 On production I get error in this line:
     if (!type.IsClass || type.IsAbstract ||, 
error is :Object reference not set to an instance of an object.

**Comment by Joniff** [comment]
That's really useful, as it matches what someone else has posted on stackoverflow.

This [stackoverflow answer ](http://stackoverflow.com/a/29379834/3782388) did suggest that an assembly could have a type which is Null. Which is what you seem to have on production. Though like me, I don't have such a thing existing in my dev version - even when I compile for release.

I have also seen that in some situations, not every assembly is loaded and have also written some extra code to check every possible assembly like this [stackoverflow answer](http://stackoverflow.com/a/10947832/3782388)

I'm currently just checking over this new code (fixing your bug), hasn't caused new bugs, otherwise I will be releasing a new version very soon. Nuget does take a few hours to index new nuget releases but the Umbraco package should be good to go very soon.

I'll post here when I've completed the new version.

Thank you again for this, I know it can be frustrating to have code work in one environment not in other.

**Comment by Joniff** [comment]
Have released a new version 1.0.4

Nuget: 
   [Terratype](https://www.nuget.org/packages/Terratype/1.0.4)
   [GoogleMapsV3](https://www.nuget.org/packages/Terratype.GoogleMapsV3/1.0.4)

Umbraco Package:
   [Terratype + GoogleMapsV3](https://our.umbraco.org/projects/backoffice-extensions/terratype/)

Hopefully this should fix your issue.

**Comment by MeggyPoe** [comment]
Thank you very much for checking and fixing this so fast. I just installed it through back office and it's working now!
I wish Umbraco moderators approved my account already I posted question there as reply to question I shared in my post week ago but they still didn't verify me. (So when they do I'll delete it since you responded here.)

Btw I really like new features you added in Terratype, was using your AngularGoogleMaps on my older projects.
